syne_tune/optimizer/schedulers/multiobjective/multi_objective_searcher.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+import logging
+from typing import Optional, List
+
+from syne_tune.optimizer.schedulers.searchers.searcher import (
+    BaseSearcher,
+    impute_points_to_evaluate,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+class BaseMultiObjectiveSearcher(BaseSearcher):
+    """
+    Base class for searchers that optimize multiple objectives simultaneously.
```

Comment:
Maybe one could be more specific here. For example, constrained BO makes use of multiple objectives, but optimizes only one of them. Maybe add something like ", where not one of the metrics is the main objective, but we rather aim to find a trade-off`.